src/gadgets/padded_shuffle.rs

```diff
@@ -19,17 +18,16 @@ pub fn fill_cs<CS: ConstraintSystem>(
     let mut values = Vec::with_capacity(pad_count);
 
     for _ in 0..pad_count {
-        // We need three independent variables constrained to be zeroes.
-        // We can do that with a single multiplier and two linear constraints for the inputs only.
-        // The multiplication constraint is enough to ensure that the third wire is also zero.
-        let (q, a, t) = cs.multiply(Scalar::zero().into(), Scalar::zero().into());
-        let assignment = Some(Value::zero());
-        values.push(AllocatedValue {
-            q,
-            a,
-            t,
-            assignment,
-        });
+        // Make an allocated value whose fields are all zero.
+        // Note: We could also create the 3 allocated variables using one multiplier
+        // (since the output multiplier is also zero),
+        // but instead we use the `Value` API for clarity (uses two multipliers).
```

Comment:
Does it really make a big difference in clarity, since we have to do `cs.constrain()` over each field anyways?

Reply:
Also, since we are going to remove `tag`, the code "constrain q=0, a=0, and then do t=q*a" will just become "constrain q=0, a=0" from the same multiplier.

Reply:
it does not make a huge difference - I just figured that since the `Value::allocate()` shortcut was available, we might as well use it. (In the 2-field change to cloak, these two paths will be equivalent anyway!)

Reply:
OTOH, if we are going to have efficient one-by-one variable allocation w/o wasting multipliers, then the straightforward code that you have already would also be the efficient, so lets keep it as it.